metecho/api/jobs.py

```diff
@@ -116,19 +115,20 @@ def _create_branches_on_github(
     if not task:
         return epic_branch_name
 
-    # Make task branch, with latest from task:
+    # Make task branch, with custom commit or falling back to latest from epic:
     task.refresh_from_db()
     if task.branch_name:
         return task.branch_name
 
+    latest_sha = task_sha or repository.branch(epic_branch_name).latest_sha()
```

Comment:
I wonder if these calls should be moved inside `with creating_gh_branch`, since they are reaching out to the GH api and part of the general "creating a branch" process.